erpnext/stock/get_item_details.py

```diff
@@ -1364,11 +1364,16 @@ def get_price_list_currency_and_exchange_rate(args):
 
 
 @frappe.whitelist()
-def get_default_bom(item_code=None):
+def get_default_bom(item_code=None, company=None):
 	def _get_bom(item):
-		bom = frappe.get_all(
-			"BOM", dict(item=item, is_active=True, is_default=True, docstatus=1), limit=1
-		)
+		if company:
+			bom = frappe.get_all(
+				"BOM", dict(item=item, is_active=True, company=company, docstatus=1), limit=1
+			)
```

Comment:
If there are multiple BOMs and one of them is default (one of them will always be default too), that should be fetched instead. You might need to process this further to get that info.